src/app/return-credentials/gst-return/gst-return.component.html

```diff
@@ -2,95 +2,145 @@
     <div class="card-header card-header-style">GST</div>
     <div class="card-block">
-      <div formArrayName="returnForms"
-           *ngFor="let returnForm of gstReturnForm.get('returnForms').controls; let i = index;">
-        <h3 class="p-3">GST Return Credentials</h3>
-        <div [formGroupName]="i">
-          <div class=".container-fluid p-1">
-            <div class="row">
-              <label for="gstFlatNo" class="col-sm-2 col-form-label">Flat No.</label>
-              <div class="col-sm-4">
-                <input type="text"
-                      id="gstFlatNo"
-                      class="form-control"
-                      name="gstFlatNo"
-                      placeholder="Flat No."
-                      formControlName="gstFlatNo"/>
-              </div>
-              <label for="gstArea" class="col-sm-2 col-form-label">Area</label>
-              <div class="col-sm-4">
-                <input type="text"
-                      id="gstArea"
-                      class="form-control"
-                      name="gstArea"
-                      placeholder="Area"
-                      formControlName="gstArea"/>
+      <div [formGroup]="gstReturnForm">
+        <div formArrayName="returnForms"
+             *ngFor="let returnForm of gstReturnForm.get('returnForms').controls; let i = index">
+          <h3 class="p-3">{{i+1}}. GST Return Credentials</h3>
+          <div [formGroupName]="i">
+            <div class=".container-fluid p-1">
+              <div class="row">
+                <label [attr.for]="'gstFlatNo'+i" class="col-sm-2 col-form-label">Flat No.</label>
+                <div class="col-sm-4">
+                  <input type="text"
+                         [id]="'gstFlatNo'+i"
+                         class="form-control"
+                         name="gstFlatNo"
+                         placeholder="Flat No."
+                         formControlName="gstFlatNo"
+                         [ngClass]="{ 'is-invalid': (returnForm.controls.gstFlatNo.touched || returnForm.controls.gstFlatNo.dirty) &&
+                         returnForm.controls.gstFlatNo.invalid}"/>
+                  <div *ngIf="(returnForm.controls.gstFlatNo.touched || returnForm.controls.gstFlatNo.dirty) &&
+                  returnForm.controls.gstFlatNo.errors?.required" class="invalid-input">
+                    Flat No. is required
+                  </div>
+                </div>
+                <label [attr.for]="'gstArea'+i" class="col-sm-2 col-form-label">Area</label>
+                <div class="col-sm-4">
+                  <input type="text"
+                         [id]="'gstArea'+i"
+                         class="form-control"
+                         name="gstArea"
+                         placeholder="Area"
+                         formControlName="gstArea"
+                         [ngClass]="{ 'is-invalid': (returnForm.controls.gstArea.touched || returnForm.controls.gstArea.dirty) &&
```

Comment:
can we separate out this validation and pass 'name' in that function that does that ?
it will be more cleaner

Reply:
I tried that but it is much tougher to do that when formarray is involved